janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -591,13 +592,16 @@ private String getDeletionScript(KeyInformation.IndexRetriever informations, Str
                 case SET:
                 case LIST:
                     String jsValue = convertToJsType(deletion.value, compat.scriptLang(), Mapping.getMapping(keyInformation));
-                    script.append("def index = ctx._source[\"").append(deletion.field).append("\"].indexOf(").append(jsValue).append("); ctx._source[\"").append(deletion.field).append("\"].remove(index);");
+                    String index = "index" + i;
```

Comment:
Might be a little cleaner to replace with `"index" + i++` here and below and remove the standalone `i++` lines.